thirdparty/pcre2/src/sljit/sljitExecAllocator.c

```diff
@@ -152,9 +152,6 @@ static SLJIT_INLINE void apple_update_wx_flags(sljit_s32 enable_exec)
 {
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= 110000
 	pthread_jit_write_protect_np(enable_exec);
-#elif defined(__clang__)
-	if (__builtin_available(macOS 11.0, *))
-		pthread_jit_write_protect_np(enable_exec);
 #else
 #error "Must target Big Sur or newer"
```

Comment:
As of this PR cherry-picked into the 3.x branch, I am unable to compile Godot on macOS, because Godot 3.x targets macOS 10.15+, and this change breaks support for macOS < 11.0.

Reply:
> I am unable to compile Godot on macOS, because Godot 3.x targets macOS 10.15+, and this change breaks support for macOS < 11.0.

Target version should be changed to 11.0, 10.15 for arm64 existed only as the early alphas.

Reply:
Thanks, fixed in https://github.com/godotengine/godot/pull/82007